### PR TITLE
Add ability to use moi init generic-ly

### PR DESCRIPTION
**Pull request by squirrelo**
Adds ability to set host, onopen, onclose, and onerror functionality when invoking the init for moi.

### CODE REVIEW

File: moi.js [hunk 1]
```diff
@@ -57,41 +57,56 @@ var moi = new function () {
...
+    this.init = function(host, group_id, on_open, on_close, on_error) {
+        host = host || window.location.host + '/moi-ws/';
         if (!("WebSocket" in window)) {
             alert("Your browser does not appear to support websockets!");
             return;
         }
-        ws = new WebSocket(host);
-        
+        //check if we need regular or secure websocket
+        socket = 'ws://';
```
Comment (wasade) [inline moi.js]: could compress some...

`socket = window.location.protocol == "https:" ? 'wss://' : 'ws://'`;

**Comment by wasade** [comment]
Minor comment, otherwise :+1:

File: moi.js (hunk 1 above)
Comment (squirrelo) [inline moi.js]: Didn't know that was a thing. Cool!

Comment (wasade) [inline moi.js]: ternary operator. and it has its c-style syntax. so awesome

**Comment by wasade** [comment]
I restarted the build, it looked like travis errored. 

Can you add a change log mention please?

**Comment by wasade** [comment]
...it looks like we're missing a .travis.yml file, that could be the issue with the tests...

**Comment by squirrelo** [comment]
Yeh, that's probably a good thing to have.

File: moi.js [hunk 2]
```diff
@@ -57,41 +57,55 @@ var moi = new function () {
...
+     * @param {host} The URL for the websocket, minus the ws:// header, or null
+     * to use the default moi-ws.
+     * @param {group_id} A group ID to get initial data from, or null to fetch
+     * all records associated with the user.
+     * @param {on_open} Optional function for action when websocket is opened.
+     * @param {on_close} Optional function for action when websocket is closed.
+     * @param {on_error} Optional function for action when websocket errors.
      */
-    this.init = function(group_id) {
+    this.init = function(host, group_id, on_open, on_close, on_error) {
```
Comment (ElDeveloper) [inline moi.js]: group_id should still be the first argument here, otherwise this is a very non-backwards compatible change.

Comment (squirrelo) [inline moi.js]: oops, good catch. Fixed.

Comment (ElDeveloper) [inline moi.js]: Thanks!

**Comment by wasade** [comment]
@squirrelo, able to kick these tests?

**Comment by squirrelo** [comment]
Kicked, lets see how they go.

**Comment by wasade** [comment]
@squirrelo, passing but has conflicts

**Pull request merged**